.github/workflows/sub_prepare.yml

```
@@ -0,0 +1,165 @@
+# It aims at running some basic checks about the workflow run ...
+# ... and gathering some data needed for the next steps.
+
+name: Prepare
+
+on:
+  workflow_call:
+    inputs:
+      artifactBasename:
+        type: string
+        required: true
+      failOnOldRebase:
+        default: false
+        type: boolean
+        required: false
+      maxRebaseHours:
+        default: "48"
+        type: string
+        required: false
+    outputs:
+      reportFile:
+        value: ${{ jobs.Prepare.outputs.reportFile }}
+      changedFiles:
+        value: ${{ jobs.Prepare.outputs.changedFiles }}
+      changedPythonFiles:
+        value: ${{ jobs.Prepare.outputs.changedPythonFiles }}
+      changedCppFiles:
+        value: ${{ jobs.Prepare.outputs.changedCppFiles }}
+
+jobs:
+
+  Prepare:
+    env:
+      isPR: ${{ github.event_name == 'pull_request' }}
+      isPush: ${{ github.event_name == 'push' }}
+      logdir: /tmp/logs/
+      reportdir: /tmp/report/
+      reportfilename: ${{ inputs.artifactBasename }}-report.md
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    outputs:
+      reportFile: ${{ steps.Init.outputs.reportFile }}
+      changedFiles: ${{ steps.Output.outputs.changedFiles }}
+      changedPythonFiles: ${{ steps.Output.outputs.changedPythonFiles }}
+      changedCppFiles: ${{ steps.Output.outputs.changedCppFiles }}
+
+    steps:
+      - name: Make needed directories, files and initializations
+        id: Init
+        run: |
+          mkdir -p ${{ env.logdir }}
+          mkdir -p ${{ env.reportdir }}
+          commitCnt=0
+          touch ${{ env.logdir }}changedFiles.lst ${{ env.logdir }}changedCppFiles.lst ${{ env.logdir }}changedPythonFiles.lst
+          echo "reportFile=${{ env.reportfilename }}" >> $GITHUB_OUTPUT
+      - name: Determine base and head SHA in case of PR
+        if: env.isPR == 'true'
+        run: |
+          baseSha=${{ github.event.pull_request.base.sha }}
+          headSha=${{ github.event.pull_request.head.sha }}
+          echo "baseSha=$baseSha" >> $GITHUB_ENV
+          echo "headSha=$headSha" >> $GITHUB_ENV
+          echo "This CI run is performed on a Pull Request" | tee -a ${{env.reportdir}}${{ env.reportfilename }}
+          echo "Base SHA is $baseSha, Head SHA is $headSha" | tee -a ${{env.reportdir}}${{ env.reportfilename }}
+      - name: Check if PR has been recently rebased
+        if: env.isPR == 'true'
+        run: |
+          baseDate=$(curl -H "Accept: application/vnd.github+json" -H "authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" $GITHUB_API_URL/repos/$GITHUB_REPOSITORY/commits/$baseSha | jq -r '.commit.committer.date')
+          dateDiff=$(( ( $(date +%s) - $(date -d $baseDate +%s) ) / 3600 ))
+          echo "Pull request is based on a $dateDiff hour-old commit" | tee -a ${{env.reportdir}}${{ env.reportfilename }}
+          # Exit the step with appropriate code
+          if [ $dateDiff -le ${{ inputs.maxRebaseHours }} ]
```

Note to my self :)
```suggestion
          if [ $dateDiff -gt ${{ inputs.maxRebaseHours }} ]
```
Will fix when CI run of PR is over as it is used as example

---

Fixed